File: octave/euler002/euler002.m
#!/usr/bin/env octave -qf
%  
%    File:    euler002.m
%    Author:  Marvin Smith
%    Date:    4/25/2015
% 

%  Update Path
addpath('octave/common')

%   Summation
result = 0

%  Set our max value
max_value = 4000000

%  Get the Max Fib Number
Fmax = Find_Max_Fibonacci( max_value );

%  Iterate over fibonacci numbers
for n = (1:Fmax)

    %  Get the value
    value = Fibonacci(n)

    %  Check if the value is even
    if mod(value,2) == 0
        result += value;
    end

end

%  Print Result
fprintf('%d\n', result)

File: julia/euler002/euler002.jl

function Fibonacci( x, y )
        x + y
end

max_value = 4000000

result = 0

x1 = 0
x2 = 1
value = 2
count = 0
while value < max_value
    global value = Fibonacci( x1, x2 )
    global x1 = x2
    global x2 = value
    if value % 2 == 0
        global result = result + value
    end
end
println("Result: $result")